### ISSUE TITLE
no-undef: Incorrectly warns about typeof UndefinedIdentifier statements

### ISSUE THREAD

**Issue by iskitz** [eslint/eslint#1482]
:+1: for creating ESLint, love the concept of making these rules truly configurable. Unfortunately, I've run into an annoyance that may actually be a bug

| ... | ... |
| --- | --- |
| **version** | 0.9.2 |
| **rule** | [no-undef](http://eslint.org/docs/rules/no-undef.html) |
| **code** | [typeof](http://www.ecma-international.org/ecma-262/5.1/#sec-11.4.3) UndefinedIdentifier |
| **expected** | No warning |
| **observed** | `[no-undef] 'UndefinedIdentifier' is not defined.` |

ESLint throws the warning above when linting code that checks for an identifier using this pattern

``` javascript
if (typeof UndefinedIdentifier === 'undefined') {
    /* do something */
}
```

The warning also occurs for this simpler yet equally [valid](http://www.ecma-international.org/ecma-262/5.1/#sec-11.4) statement

``` javascript
typeof UndefinedIdentifier;
```

I value ESLint's undefined warnings but prefer they only occur for code that doesn't explicitly check for just that. Disabling the `no-undef` warning or explicitly declaring each undefined identifier via `/*global ...*/` is impractical for me so I'm hoping this will be fixed . . . soon ;-)

Thanks!
